src/goto-instrument/contracts/contracts.cpp

```diff
@@ -236,10 +236,14 @@ void code_contractst::check_apply_loop_contracts(
     new_decreases_vars.push_back(new_decreases_var);
   }
 
-  // non-deterministically skip the loop if it is a do-while loop.
   // TODO: Fix loop contract issues for do/while loops.
   if(!loop_head->is_goto())
   {
+    log.error() << "Loop contracts are unsupported on do/while loops: "
+                << loop_head->source_location() << messaget::eom;
+    throw 0;
```

Comment:
Please can you throw something, anything really, rather than a `0`?  I know other bits of code do it; they really shouldn't.  A string would be fine.

Reply:
I agree that this is ugly, but I think should we make a global pass and remove it from all of the contracts code (and many other modules that `throw 0`) in its own PR. What do you think?

Reply:
If @martin-cs agrees, please file an issue to track this.

Reply:
I will accept "that is the standard for this area of the code" and a good faith 'at some point' promise to do something about it.

I ... would be a little more reluctant to do it more widely because not all of them should be `throw`s; some should be `DATA_INVARIANTS`, some should be `CHECK_RETURN`, etc.

( For those who might be following who are new to this -- `throw` should only be used for things that are possible with incorrect user input, if it isn't possible (well, shouldn't be...) then it should be one of the `INVARIANTS` depending on why it shouldn't happen. )

Reply:
In this commit I don't treat this an internal invariant failure (like violation of some precondition for a CBMC routine), but treat it as a user-side error -- the user trying to use an unsupported feature.

But I agree that this could be triggered either way and would be happy to change this to an invariant failure. What do you suggest?

Reply:
@jimgrundy An issue would even better than a good faith promise!

Reply:
Opened #6468 to track this.